src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -35,95 +33,1003 @@ const ActorInit En_Sw_InitVars = {
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->sphs);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0xE), &D_80B0F074);
+    this->actor.scale.x = 0.02f;
+    if (((thisx->params & 0xE000) >> 0xD) == 0) {
+        this->actor.world.rot.x = 0;
+        this->actor.world.rot.z = 0;
+        thisx->shape.rot = this->actor.world.rot;
+        this->unk_484.y = this->actor.world.pos.y;
+        this->unk_484.x = this->actor.world.pos.x + (Math_SinS(this->actor.world.rot.y) * -60.0f);
+        this->unk_484.z = this->actor.world.pos.z + (Math_CosS(this->actor.world.rot.y) * -60.0f);
+        func_80B0DFFC(this, globalCtx);
+        this->actor.home.pos = this->actor.world.pos;
+    } else {
+        this->unk_370.x = Math_SinS(thisx->shape.rot.y + 0x4000);
+        this->unk_370.y = 0.0f;
+        this->unk_370.z = Math_CosS(thisx->shape.rot.y + 0x4000);
+        this->unk_364.x = 0.0f;
+        this->unk_364.y = 1.0f;
+        this->unk_364.z = 0.0f;
+        this->unk_37C.x = Math_SinS(thisx->shape.rot.y);
+        this->unk_37C.y = 0.0f;
+        this->unk_37C.z = Math_CosS(thisx->shape.rot.y);
+        func_80B0C0CC(this, globalCtx, 1);
+    }
+
+    if (((thisx->params & 0xE000) >> 0xD) >= 3) {
+        Audio_PlaySoundGeneral(NA_SE_SY_CORRECT_CHIME, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    }
+
+    switch ((thisx->params & 0xE000) >> 0xD) {
+        case 3:
+        case 4:
+            this->unk_360 = 1;
+            this->actor.velocity.y = 8.0f;
+            this->actor.speedXZ = 4.0f;
+            this->actor.gravity = -1.0f;
+        case 2:
+            this->actor.scale.x = 0.0f;
+        case 1:
+            this->collider.elements[0].info.toucher.damage *= 2;
+            this->actor.naviEnemyId = 0x20;
+            this->actor.colChkInfo.health *= 2;
+            this->actor.flags &= ~1;
+            break;
+        default:
+            Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_ENEMY);
+            this->actor.naviEnemyId = 0x1F;
+            break;
+    }
+
+    this->unk_38E = Rand_S16Offset(0xF, 0x1E);
+    Actor_SetScale(&this->actor, this->actor.scale.x);
+    this->actor.home.pos = this->actor.world.pos;
+    thisx->shape.rot = this->actor.world.rot;
+
+    if (((thisx->params & 0xE000) >> 0xD) >= 3) {
+        this->unk_38C = 0x28;
+        this->unk_394 = 1;
+        this->actionFunc = func_80B0D364;
+    } else if (((thisx->params & 0xE000) >> 0xD) == 0) {
+        this->actionFunc = func_80B0E5E0;
+    } else {
+        this->actionFunc = func_80B0D590;
+    }
+}
+
+void EnSw_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnSw* this = THIS;
+
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+s32 func_80B0C9F0(EnSw* this, GlobalContext* globalCtx) {
+    s32 phi_v1;
+
+    phi_v1 = false;
+    if (this->actor.xyzDistToPlayerSq < 160000.0f && ((this->actor.params & 0xE000) >> 0xD) == 0 &&
```

Comment:
```suggestion
    s32 phi_v1 = false;
    
    if (this->actor.xyzDistToPlayerSq < 160000.0f && ((this->actor.params & 0xE000) >> 0xD) == 0 &&
```

Reply:
It might not match, but it makes sense to convert `160000.0f` to `SQ(400.0f)`